plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/MavenDependencyVersionsFetcherFile.kt

```diff
@@ -0,0 +1,35 @@
+package de.fayard.refreshVersions.core.internal
+
+import de.fayard.refreshVersions.core.ModuleId
+import java.io.File
+import java.io.FileNotFoundException
+
+internal class MavenDependencyVersionsFetcherFile(
+    moduleId: ModuleId,
+    repoUrl: String
+) : MavenDependencyVersionsFetcher(
+    moduleId = moduleId,
+    repoUrl = repoUrl
+) {
+    private val repoDir = File(repoUrl.substringAfter("file:"))
+
+    init {
+        require(repoUrl.endsWith('/'))
+    }
+
+    override suspend fun getXmlMetadataOrNull(): String? = try {
+        val targetDir = repoDir.resolve("${moduleId.group!!.replace('.', '/')}/${moduleId.name}")
+        val fileNames: Array<String>? = targetDir.list()
```

Comment:
I suggest a quick return here to avoid using `?.` everywhere
`targetDir.list() ?: return null`

Reply:
applied that suggestion